src/main/java/Main.java

```diff
@@ -0,0 +1,186 @@
+import spark.ModelAndView;
+import spark.Session;
+import spark.Spark;
+import spark.template.mustache.MustacheTemplateEngine;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by EddyJ on 7/13/16.
+ */
+public class Main {
+    static int pageNumber = 1;
+    static int restaurantLimitPerPage = 5;
+    static int id = 0;
+    static List<Restaurant> restaurants = new ArrayList<>();
+    static Map<String, User> users = new HashMap<>();
+    public static final String SESSION_USERNAME = "userName";
+
+    public static void main(String[] args) {
+
+        Spark.init();
+
+        Spark.get("/", (request, response) -> {
+            HashMap m = new HashMap();
+            Session session = request.session();
+            String userName = session.attribute(SESSION_USERNAME);
+
+            List<Restaurant> userRestaurants = new ArrayList();
+            List<Restaurant> totalRestaurants = new ArrayList();
+            List<Restaurant> currentPageList = new ArrayList();
+
+            for (Restaurant restaurant: restaurants){
+                if(restaurant.getAuthor().equals(userName)){
+                    userRestaurants.add(restaurant);
+                }
+                else{
+                    totalRestaurants.add(restaurant);
+                }
+            }
+
+            String pageString = request.queryParams("pageNumber");
+            
+            if(totalRestaurants.size() <= restaurantLimitPerPage){
+                for (Restaurant restaurant : totalRestaurants){
+                    currentPageList.add(restaurant);
+                }
+            }
+            else{
+                int maxIndex = (pageNumber * restaurantLimitPerPage - 1);
+                int minIdex = (maxIndex - restaurantLimitPerPage + 1);
+                for(int i = minIdex; i < maxIndex; i++)
+                    if(i < restaurants.size()){
+                        currentPageList.add(restaurants.get(i));
+                    }
+                pageNumber++;
+            }
+
+
+            m.put("pageNumber", pageNumber);
+            m.put("userRestaurants", userRestaurants);
+            m.put("currentPageList", currentPageList);
+            m.put("user", userName);
+
+            return new ModelAndView(m, "home.html");
+        }, new MustacheTemplateEngine());
+
+        Spark.post("/create-user" , (request, response) -> {
+            String name = request.queryParams("loginName");
+            User user = users.get(name);
+
+            if( user == null && !(users.containsKey(user))){
```

Comment:
If `user == null` then why would you ever test whether or not `users` contains the key for user.  This line doesn't actually work since the `containsKey` is expecting a string not a user object so it will look for the string representation of the user object.